### DISCUSSION TITLE
1.7.1: in daily consolidation, values of next day are included

### DISCUSSION THREAD

**Discussion by piet66-peb** [oetiker/rrdtool-1.x#1216] [Q&A]
There's something I don't understand.
I have tried to create some charts with consolidated data, but the results are not what I expected.

Checking the xml file, I found that for a day's consolidation, the function uses values from 01:00+x local time to 01:00 of the following day, in the case of daylight saving time, from 02:00+x local time to 02:00 of the following day. In other words: consolidation interval is the utc day, not the local day.

The time stamp of the consolidated value is then 01:00 the next day, which is very irritating.

I would expect it to take all values from 00:00+x to 24:00 local time, if midnight belongs to the previous day.

Or have I misunderstood something? Is there a particular reason for this behavior?

And another strange thing:
In consolidation, a week always goes from Thursday 01:00+x to Thursday 01:00 next week.
I would expect: from Monday 00:00+x to Monday 00:00 next week

**Comment by oetiker** [comment]
Internally rrdtool is always UTC aligned so if you consolidate 'by day' then the day is a UTC day ... when rendering a chart, then the grid of the chart will be in local time. If you want to be able to create day by day reports in local time, then you must stick with consolidation on the hour level.

**Comment by piet66-peb** [comment]
Thanks for the quick reply.

Unfortunately, I don't like it because I wonder what consolidation to utc level is worth to the user. 
I can't see any benefit at the moment.

**Reply by oetiker** [reply]
if you can not see a benefit in consolidating to 24h intervals at utc midnight, then you should not use that consolidation level ... the reason that all is utc centric is that as soon as you go across timezones this is the only way to stay sane AND consolidate ... 

if you want to do analysis taking the local 'day' into account you need 1h resolution data.
